Dockerfile

```diff
@@ -0,0 +1,36 @@
+# Use the Rust image based on Debian Bookworm as the base image
+FROM rust:slim-bookworm as builder
+
+# Install dependencies
+RUN apt-get update && apt-get install -y \
+    libssl-dev \
+    pkg-config \
+    clang \
+    git \
+    && rm -rf /var/lib/apt/lists/*
+
+# Clone the repository
+WORKDIR /app
+RUN git clone https://github.com/sabledb-io/sabledb.git .
+WORKDIR /app
+
+# Build the project
+RUN cargo build --release
+
+# Copy the build artifact to a slim image
+FROM debian:bookworm-slim
+
+# Install necessary runtime dependencies
+RUN apt-get update && apt-get install -y \
+    libssl-dev \
+    libclang-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+# Copy the built binary from the builder stage
+COPY --from=builder /app/target/release/sabledb /usr/local/bin/sabledb
```

Comment:
I would actually install it under `/opt/sabledb`
With the following directory structure:

```
/opt/sabledb/bin
/opt/sabledb/log
/opt/sabledb/etc
/opt/sabledb/db
```

Copy the following binaries:

```
target/release/sabledb -> /opt/sabledb/bin
target/release/sdb-cli -> /opt/sabledb/bin
target/release/sb ->  /opt/sabledb/bin
```

In addition, generate `sabledb.ini` file under `/opt/sabledb/etc`:

```bash
cat << EOF >> /opt/sabledb/etc/sabledb.ini
logdir = "/opt/sabledb/log"
listen_ip = "0.0.0.0"
db_path = "/opt/sabledb/db"
config_dir = "/opt/sabledb/etc"
EOF
```

And change the entry point to be:

```
["/opt/sabledb/bin/sabledb", "/opt/sabledb/etc/sabledb.ini"]
```

Reply:
Actually, I am happy with these changes. I am going to merge this one PR and I am going to modify it a bit later by adding a template `server.docker.ini` file in the repository to reduce the amount of work the user needs to do later on

Reply:
Sure, sounds good.